tamr_toolbox/data_io/dataframe.py

```diff
@@ -317,13 +317,40 @@ def _check_nonnull_columns(
     return ValidationCheck(passed, failed_checks_dict)
 
 
+def _check_custom(
+    df: "pandas.DataFrame",
+    columns_to_check: Optional[List[str]],
+    check_function: Callable[[Any], bool],
+) -> ValidationCheck:
+    """
+    Args:
+        df: Dataframe
+        columns_to_check: columns on which check_function will be applied
+        check_function: function applied on columns_to_check
+
+    Returns:
+        ValidationCheck object, with bool for whether all checks passed and dict of failing columns
+    """
+    failed_checks_dict = defaultdict(list)
+
+    df1 = df[columns_to_check].applymap(check_function)
+    for col in columns_to_check:
+        if not df1[col].all():
+            LOGGER.warning(f"column {col} failed custom check {check_function.__name__}")
+            failed_checks_dict[f"failed custom check {check_function.__name__}"].append(col)
+
+    passed = len(failed_checks_dict) == 0
+    return ValidationCheck(passed, failed_checks_dict)
+
+
 def validate(
     df: "pandas.DataFrame",
     *,
     raise_error: bool = True,
     require_present_columns: Optional[List[str]] = None,
     require_unique_columns: Optional[List[str]] = None,
     require_nonnull_columns: Optional[List[str]] = None,
+    custom_checks: Iterable[Tuple[Callable[[Any], bool], List[str]]] = None,
```

Comment:
```suggestion
    custom_checks: Iterable[Tuple[Callable[[Any], bool], List[str]]] = tuple(),
```
What do you think of making this an empty tuple?  `()` is also a valid way of writing it but this might be clearer.